integrated_channels/sap_success_factors/client.py

```diff
@@ -0,0 +1,54 @@
+"""
+Client for connecting to SAP SuccessFactors.
+"""
+from __future__ import absolute_import, unicode_literals
+
+import requests
+import slumber
+
+from integrated_channels.sap_success_factors.models import SAPSuccessFactorsGlobalConfiguration
+
+
+class SAPSuccessFactorsAPIClient(slumber.API):
+
+    @classmethod
+    def get_oauth_access_token(cls, url_base, client_id, client_secret, company_id, user_id):
+        """ Retrieves OAuth 2.0 access token using the client credentials grant.
+
+        Args:
+            url_base (str): Oauth2 access token endpoint
+            client_id (str): client ID
+            client_secret (str): client secret
+            company_id (str): SAP company ID
+            user_id (str): SAP user ID
+
+        Returns:
+            tuple: Tuple containing access token string and expiration datetime.
+        """
+
+        global_sap_config = SAPSuccessFactorsGlobalConfiguration.current()
+        url = url_base + global_sap_config.oauth_api_path
+
+        response = requests.post(
+            url,
+            data={
+                'grant_type': 'client_credentials',
+                'scope': {
+                    'userId': user_id,
+                    'companyId': company_id,
+                    'userType': 'user',
+                    'resourceType': 'learning_public_api',
+                }
+            },
+            auth=(client_id, client_secret)
+        )
+
+        data = response.json()
+
+        try:
+            access_token = data['access_token']
+            expires_at = data['expiresIn']
```

Comment:
Do we want to try to convert this into a (preferably timezone-aware) datetime object based on the documented SAP response format?